src/include/test/unit_tdb_matrix.cc

```diff
@@ -271,3 +271,49 @@ TEST_CASE("tdb_matrix: MatrixBase template parameter", "[tdb_matrix]") {
     }
   }
 }
+
+TEST_CASE("tdb_matrix: empty matrix", "[tdb_matrix]") {
+  tiledb::Context ctx;
+  std::string tmp_matrix_uri =
+      (std::filesystem::temp_directory_path() / "tmp_tdb_matrix").string();
+  size_t matrix_dimension{128};
+  int32_t matrix_domain{1000};
+  int32_t tile_extent{100};
+
+  tiledb::VFS vfs(ctx);
+  if (vfs.is_dir(tmp_matrix_uri)) {
+    vfs.remove_dir(tmp_matrix_uri);
+  }
+
+  create_empty_for_matrix<float, stdx::layout_left>(
+      ctx,
+      tmp_matrix_uri,
+      matrix_dimension,
+      matrix_domain,
+      matrix_dimension,
+      tile_extent);
+
+  SECTION("empty") {
+    auto X =
+        tdbColMajorMatrix<float>(ctx, tmp_matrix_uri, 0, 0, 0, 0, 10000, 0);
+    X.load();
+    CHECK(X.num_cols() == 0);
+    CHECK(num_vectors(X) == 0);
+    CHECK(X.num_rows() == 0);
+    CHECK(dimension(X) == 0);
+  }
+  SECTION("filled") {
+    auto X = tdbColMajorMatrix<float>(ctx, tmp_matrix_uri);
+    X.load();
+    CHECK(X.num_cols() == matrix_domain);
+    CHECK(num_vectors(X) == matrix_domain);
+    CHECK(X.num_rows() == matrix_dimension);
+    CHECK(dimension(X) == matrix_dimension);
+
+    auto Y = tdbColMajorMatrix<float>(std::move(X));
+    CHECK(Y.num_cols() == X.num_cols());
```

Comment:
Sure, done, though it shouldn't affect things here because all these checks for for variables initialized in the constructor. That said, it is a good to verify that `Y.load();` doesn't change anything, so I updated to that.